### PR TITLE
Don't allocate in ProcVFS.getOpenFdCount

### PR DESCRIPTION
**Pull request by nemanja-boric-sociomantic**
`ProcVFS.openFdCount` was using tango's FilePath facilities
which are iduping the name entries for the `FileInfo` structure.
Since we're interested just in the file descriptor count, the
implementation is changed to opendir/readdir/closedir alternative.

### CODE REVIEW

File: src/ocean/sys/stats/linux/ProcVFS.d [hunk 1]
```diff
@@ -78,9 +80,38 @@ public int getOpenFdCount ()
...
+
+    scope (exit)
     {
-        count++;
+        // ignore EBADF
+        closedir(dir);
+    }
+
+    /*
+        From http://man7.org/linux/man-pages/man3/readdir_r.3.html
```
Comment (gavin-norman-sociomantic) [inline src/ocean/sys/stats/linux/ProcVFS.d]: Why this comment about thread safety?

File: src/ocean/sys/stats/linux/ProcVFS.d [hunk 2]
```diff
@@ -78,9 +80,38 @@ public int getOpenFdCount ()
...
+        thread-safe. Therefore, the use of readdir_r() is generally
+        unnecessary in multithreaded programs. In cases where multiple
+        threads must read from the same directory stream, using readdir(3)
+        with external synchronization is still preferable to the use of
+        readdir_r(), for the reasons given in the points above.
+    */
+    dirent* dp;
+    while ((dp = readdir(dir)) !is null)
+    {
+        if (strcmp(dp.d_name.ptr, "..".ptr) != 0 &&
```
Comment (gavin-norman-sociomantic) [inline src/ocean/sys/stats/linux/ProcVFS.d]: Tiny point: I'm wondering if `StringC.toDString(dp.d_name.ptr) == ".."` would be more idiomatic and slightly easier to read. (Or was there a more modern / favoured alternative to `ocean.text.util.StringC`?)

Comment (nemanja-boric-sociomantic) [reply src/ocean/sys/stats/linux/ProcVFS.d]: Fair enough, I'll replace it with `toDString`.

File: src/ocean/sys/stats/linux/ProcVFS.d (hunk 1 above)
Comment (nemanja-boric-sociomantic) [reply src/ocean/sys/stats/linux/ProcVFS.d]: This is a note about `readdir` vs `readdir_r`. Normally we should use `*_r` methods, however `readdir` is a notable exception. I'll expand the comment.

**Comment by nemanja-boric-sociomantic** [comment]
Updated.

File: src/ocean/sys/stats/linux/ProcVFS.d [hunk 3]
```diff
@@ -20,6 +20,8 @@ import ocean.transition;
 import Path = ocean.io.Path;
 import ocean.sys.ErrnoException;
 import core.sys.posix.stdio;
+import core.sys.posix.dirent;
+import core.stdc.string;
```
Comment (gavin-norman-sociomantic) [inline src/ocean/sys/stats/linux/ProcVFS.d]: Unused import now?

Comment (nemanja-boric-sociomantic) [reply src/ocean/sys/stats/linux/ProcVFS.d]: Ugh.. :+1: updated.

**Review by mihails-strasuns — APPROVED** [review]

**Pull request merged**